File: src/array/array_java/Ch6_ReverseAnArray.java
package array.array_java;

import java.util.Scanner;

public class Ch6_ReverseAnArray {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int[] arr = new int[n];
        System.out.println("Print Array");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = scn.nextInt();
        }

        int i = 0;
        int j = arr.length - 1;
        while (i <= j) {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
            i++;
            j--;
        }
        System.out.println("Reverse Array");

        for (int k = 0; k < arr.length; k++) {
            System.out.print(arr[k] + " ");
        }
    }
}

File: src/array/array_kotlin/Ch6_ReverseAnArray.kt
package array.array_kotlin

import java.util.*

fun main() {
    reverseAnArray()
}

fun reverseAnArray() {
    val scn = Scanner(System.`in`)
    val n = scn.nextInt()
    val arr = IntArray(n)
    println("Print Array")
    for (i in arr.indices) {
        arr[i] = scn.nextInt()
    }

    var i = 0
    var j = arr.size - 1
    while (i <= j) {
        val temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp
        i++
        j--
    }
    println("Reverse Array")

    for (k in arr.indices) {
        print(arr[k].toString() + " ")
    }
}
